src/js/utils/SchemaFormUtil.js

```diff
@@ -166,6 +166,129 @@ function processValue(value, valueType) {
   return value;
 }
 
+/**
+ * Recursive function to flatten the 'group' types
+ * in a definiton.
+ *
+ * @param {object} definition - The input definition
+ * @returns {object} The flattened definition
+ */
+function unnestGroupsInDefinition(definition) {
+  let defClone = Object.assign({}, definition);
+  if (defClone.properties == null) {
+    return defClone;
+  }
+
+  // Make sure we are operating on a cloned properties
+  defClone.properties = Object.assign({}, defClone.properties);
```

Comment:
Let's move the cloning into it's own util - thinking about a general "deep" clone util.